### PR TITLE
Some simple Option / Result / iterator pattern simplifications

### PR DESCRIPTION
**Pull request by huitseeker**
Makes a couple of places more readable. Opinions on these refactorings
loosely held, happy to accomodate review desiderata.

---
name: Pull Request
about: Pull Request checklist
title: ''
labels: ''
assignees: ''

---
If your PR is a work in progress, please feel free to create it and include a [WIP] tag in the PR name. We encourage everyone to PR early and often so that other developers know what you're working on.

Before submitting your PR for final review, please ensure that it:

* Includes a proper description of what problems the PR addresses, as well as a detailed explanation as to what it changes
* Explains whether/how the change is consensus breaking or breaks existing client functionality
* Contains unit tests exercising new/changed functionality
* Fully considers the potential impact of the change on other parts of the system
* Describes how you've tested the change (e.g. against Floonet, etc)
* Updates any documentation that's affected by the PR

### CODE REVIEW

File: api/src/types.rs [hunk 1]
```diff
@@ -334,10 +334,10 @@ impl OutputPrintable {
...
 
 	pub fn range_proof(&self) -> Result<pedersen::RangeProof, ser::Error> {
-		let proof_str = match self.proof.clone() {
-			Some(p) => p,
-			None => return Err(ser::Error::HexError(format!("output range_proof missing"))),
-		};
+		let proof_str = self
+			.proof
+			.clone()
+			.ok_or_else(|| ser::Error::HexError(format!("output range_proof missing")))?;
```
Comment (quentinlesceller) [inline api/src/types.rs]: Not sure why, but in that specific case, I find the previous more readable.

File: api/src/web.rs [hunk 2]
```diff
@@ -105,10 +105,7 @@ impl QueryParams {
 	}
 
 	pub fn get(&self, name: &str) -> Option<&String> {
-		match self.params.get(name) {
-			None => None,
-			Some(v) => v.first(),
-		}
+		self.params.get(name).and_then(|v| v.first())
```
Comment (quentinlesceller) [inline api/src/web.rs]: 👍

File: core/src/libtx/proof.rs [hunk 3]
```diff
@@ -228,9 +225,10 @@ where
 		let id = Identifier::from_serialized_path(depth, &msg[4..]);
 
 		let commit_exp = self.keychain.commit(amount, &id, &switch)?;
-		match commit == &commit_exp {
-			true => Ok(Some((id, switch))),
-			false => Ok(None),
+		if commit == &commit_exp {
+			Ok(Some((id, switch)))
+		} else {
+			Ok(None)
```
Comment (quentinlesceller) [inline core/src/libtx/proof.rs]: This commit is also valid for the next occurrences of match replacement for if else.
This is valid but I'm wondering if it's more readable than the previous match case?

File: pool/src/pool.rs [hunk 4]
```diff
@@ -403,11 +403,7 @@ impl Pool {
 		// Oldest (based on pool insertion time) will then be prioritized.
 		tx_buckets.sort_unstable_by_key(|x| (Reverse(x.fee_to_weight), x.age_idx));
 
-		tx_buckets
-			.into_iter()
-			.map(|x| x.raw_txs)
-			.flatten()
-			.collect()
+		tx_buckets.into_iter().flat_map(|x| x.raw_txs).collect()
```
Comment (quentinlesceller) [inline pool/src/pool.rs]: 👍

File: core/src/libtx/proof.rs (hunk 3 above)
Comment (antiochp) [reply core/src/libtx/proof.rs]: I think clippy actually prefers you use if/else on a bool - 
https://users.rust-lang.org/t/is-it-bad-style-to-match-a-bool/14359

**Comment by antiochp** [comment]
Big :+1: on "Opinions on these refactorings loosely held" by the way. Really appreciate this!

**Pull request merged**